File: Core/Entities/SearchParameters.cs
namespace Core.Entities;

public class SearchParameters
{
    // The folders to process
    public string[] Folders { get; set; } = [];

    // The category of the files we will be processing
    public FileSearchType? FileSearchType { get; set; }

    // The degree of similarity between images (only for images)
    public int? DegreeOfSimilarity { get; set; }

    // Should we include subfolders or not
    public bool IncludeSubfolders { get; set; }

    // The minimum size for files to include in search
    public long? MinSize { get; set; }

    // The maximum size for files to include
    public long? MaxSize { get; set; }

    // The file types to include. With this excluded files types are disabled
    public string[] IncludedFileTypes { get; set; } = [];

    // The file types to exclude. Only work if no file types are included
    public string[] ExcludedFileTypes { get; set; } = [];
}

File: web/src/app/shared/models/search-parameters.ts
import { FileSearchType } from './file-search-type';
import { PerceptualHashAlgorithm } from './perceptual-hash-algorithm';

export class SearchParameters {
  folders: string[] = [];
  fileSearchType: FileSearchType | null = null;
  degreeOfSimilarity: number | null = null;
  perceptualHashAlgorithm: PerceptualHashAlgorithm | null = null;
  includeSubFolders: boolean = true;
  minSize: number | null = null;
  maxSize: number | null = null;
  includedFileTypes: string[] = [];
  excludedFileTypes: string[] = [];
}
